src/app/api/tasks/tasks.controller.ts

```diff
@@ -50,10 +50,10 @@ export const deleteTask = async (req: NextRequest, { params: { id } }: IdParams)
   return new NextResponse(null, { status: httpStatus.NO_CONTENT })
 }
 
-export const completeTask = async (req: NextRequest, { params: { id } }: IdParams) => {
+export const clientUpdateTask = async (req: NextRequest, { params: { id } }: IdParams) => {
   const user = await authenticate(req)
-
+  const workflowStateId = req.nextUrl.searchParams.get('workflowStateId') ?? undefined
```

Comment:
I think we could use the default null return from `#nextUrl.searchParams.get`, and accomodate `clientUpdateTask` for this.